### PR TITLE
Renew daemon

### PR DESCRIPTION
**Pull request by maraino**
### Description
Add support for a renew daemon that will renew periodically a certificate.

### CODE REVIEW

File: command/ca/renew.go [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+		Action: command.ActionFunc(renewCertificateAction),
+		Usage:  "renew a valid certificate",
+		UsageText: `**step ca renew** <crt-file> <key-file>
+		[**--ca-url**=<uri>] [**--root**=<file>]
+		[**--out**=<file>] [**--expires-in**=<duration>] [**--force**]`,
+		Description: `
+**step ca renew** command renews the given certificates on the certificate
+authority and writes the new certificate to disk either overwriting <crt-file>
+or using a new file if the **--out**=<file> flag is used.
+å
```
Comment (dopey) [inline command/ca/renew.go]: weird character. I'll do a grammatical once over once I'm through reading the rest of the commit.

Comment (maraino) [reply command/ca/renew.go]: Fixed.

File: command/ca/renew.go [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+
+Renew the certificate and reload nginx:
+'''
+$ step ca renew --daemon --exec "nginx -s reload" internal.crt internal.key 
+'''
+
+Renew the certificate and convert it to DER:
+'''
+$ step ca renew --daemon --renew-period 16h \
+  --exec "step certificate format --force --out internal.der internal.crt" \
```
Comment (dopey) [inline command/ca/renew.go]: Why use --force if --out is different from the original file?

Comment (maraino) [reply command/ca/renew.go]: because this is going to run periodically multiple times

**Pull request merged**